src/coreclr/jit/lowerxarch.cpp

```diff
@@ -4949,7 +4949,7 @@ void Lowering::ContainCheckCompare(GenTreeOp* cmp)
         // we can treat the MemoryOp as contained.
         if (op1Type == op2Type)
         {
-            if (IsContainableMemoryOp(op1))
+            if (IsContainableMemoryOp(op1) && IsSafeToContainMem(cmp, op1))
```

Comment:
There looked to be a couple outer places in lowerxarch that do `IsContainableMemoryOp` followed by an `MakeSrcContained`
* https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/lowerxarch.cpp#L4726-L4729
* https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/lowerxarch.cpp#L4856-L4859
* https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/lowerxarch.cpp#L5253-L5256
* https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/lowerxarch.cpp#L5282-L5286
* https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/lowerxarch.cpp#L6290-L6298

We also have https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/lowerxarch.cpp#L5072-L5075, where its not clear if it doing `IsContainableMemoryOp` is right and perhaps `indirOpSource->isContained()` would be better?

Reply:
Thanks, added checks.

I'll probably leave that last bit as is, `IsContainableMemoryOp` is not that expensive.